### DISCUSSION TITLE
Why can't we rebind the dot character, and why is repeating inserts privileged?

### DISCUSSION THREAD

**Discussion by 7ombie** [helix-editor/helix#7411] [Q&A] [answered]
Is there a technical reason for not allowing the dot key to be rebound, or for not allowing any other binding to repeat the last insert?

I'd like to use the dot and comma keys to group selection-related stuff:

+ `.` collapse selections
+ `Alt-.` flip selections
+ `,` drop all but primary selection
+ `Alt-,` drop primary selection
+ `<` cycle primary selection backwards
+ `>` cycle primary selection forward

Note: I have other bindings for indent and dedent, so `<` and `>` are free.

I can then use the semi-colon/colon key for commands and macros.

I find it more intuitive to group similar operations on the same keys, but binding `.` is not supported, and there's no command for `repeat_last_insert`.

**Comment by the-mikedavis** [comment, marked as answer]
It's currently hard-coded but it could probably be refactored as a command: https://github.com/helix-editor/helix/issues/1488#issuecomment-1012254773

**Reply by 7ombie** [reply]
Thank you. I did search (GitHub and Google), but couldn't find anything.

**Reply by 7ombie** [reply]
@the-mikedavis - I'm aware that I've come across quite negatively over the last few days. I've generally praised Helix, and promoted it in other places. It's an awesome project, and I appreciate your patience personally with handling negative feedback. I just wanted to adapt it in ways that clashed with your broader vision for the project, and let my frustration piss me off a bit. Sorry for that, mate. All the best.
